src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,73 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private final Map<Class<?>, Object> instances = new HashMap<>();
+    private final Map<Class<?>, Class<?>> implementationsMap = new HashMap<>();
+
+    private Injector() {
+        implementationsMap.put(ProductService.class, ProductServiceImpl.class);
+        implementationsMap.put(ProductParser.class, ProductParserImpl.class);
+        implementationsMap.put(FileReaderService.class, FileReaderServiceImpl.class);
+    }
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        Class<?> clazz = findImplementation(interfaceClazz);
+        return createInstance(clazz);
+    }
+
+    private Class<?> findImplementation(Class<?> interfaceClazz) {
+        if (interfaceClazz.isInterface()) {
+            return implementationsMap.get(interfaceClazz);
+        }
+        return interfaceClazz;
+    }
+
+    private Object createInstance(Class<?> clazz) {
+        if (instances.containsKey(clazz)) {
+            return instances.get(clazz);
+        }
+
+        try {
+            Constructor<?> constructor = clazz.getDeclaredConstructor();
+            Object instance = constructor.newInstance();
+            instances.put(clazz, instance);
+            injectDependencies(instance);
+
+            return instance;
+        } catch (Exception e) {
+            throw new RuntimeException("Can't create a new instance of " + clazz.getName(), e);
```

Comment:
The error message could be more informative by including the original exception message or type. This can help in debugging the specific cause of the failure.